src/WebJobs.Script.WebHost/Metrics/LinuxContainerLegionMetricsPublisher.cs

```diff
@@ -0,0 +1,198 @@
+
+            _metricsFileManager = new LegionMetricsFileManager(metricsFilePath, fileSystem, logger, maxFileCount);
+
+            if (_standbyOptions.CurrentValue.InStandbyMode)
+            {
+                _standbyOptionsOnChangeSubscription = _standbyOptions.OnChange(o => OnStandbyOptionsChange());
+            }
+            else
+            {
+                Start();
+            }
+        }
+
+        public void Initialize()
+        {
+            _process = Process.GetCurrentProcess();
+            _initialized = true;
+        }
+
+        public void Start()
+        {
+            Initialize();
+            _processMonitorTimer = new Timer(OnProcessMonitorTimer, null, _timerStartDelay, _memorySnapshotInterval);
+            _metricsPublisherTimer = new Timer(OnFunctionMetricsPublishTimer, null, _timerStartDelay, _metricPublishInterval);
+
+            _logger.LogInformation(string.Format("Starting metrics publisher for container : {0}.", _containerName));
+        }
+
+        private void OnStandbyOptionsChange()
+        {
+            if (!_standbyOptions.CurrentValue.InStandbyMode)
+            {
+                Start();
+            }
+        }
+
+        public void AddFunctionExecutionActivity(string functionName, string invocationId, int concurrency, string executionStage, bool success, long executionTimeSpan, string executionId, DateTime eventTimeStamp, DateTime functionStartTime)
+        {
+            if (!_initialized)
+            {
+                return;
+            }
+
+            Enum.TryParse(executionStage, out FunctionExecutionStage functionExecutionStage);
+
+            FunctionActivity activity = new FunctionActivity
+            {
+                FunctionName = functionName,
+                InvocationId = invocationId,
+                Concurrency = concurrency,
+                ExecutionStage = functionExecutionStage,
+                ExecutionId = executionId,
+                IsSucceeded = success,
+                ExecutionTimeSpanInMs = executionTimeSpan,
+                EventTimeStamp = eventTimeStamp,
+                StartTime = functionStartTime
+            };
+
+            _metricsTracker.AddFunctionActivity(activity);
+        }
+
+        public void AddMemoryActivity(DateTime timeStampUtc, long data)
+        {
+            if (!_initialized)
+            {
+                return;
+            }
+
+            var memoryActivity = new MemoryActivity
+            {
+                CommitSizeInBytes = data,
+                EventTimeStamp = timeStampUtc
+            };
+
+            _metricsTracker.AddMemoryActivity(memoryActivity);
+        }
+
+        private async void OnFunctionMetricsPublishTimer(object state)
+        {
+            await OnPublishMetrics();
+        }
+
+        internal async Task OnPublishMetrics()
+        {
+            try
+            {
+                if (_metricsTracker.TryGetMetrics(out LinuxConsumptionMetrics trackedMetrics))
```

Comment:
The tracker returns aggregated metrics based on the activities that were sent to it, as well as its own aggregation intervals